lib/config/validation.ts

```diff
@@ -111,6 +113,18 @@ function isInhertConfigOption(key: string): boolean {
   return optionInherits.has(key);
 }
 
+function isRegexOrGlobOption(key: string): boolean {
+  if (!optionRegexOrGlob) {
+    optionRegexOrGlob = new Set();
+    for (const option of options) {
```

Comment:
we should extract a single function to init, so we don't need to iterate multiple times

Reply:
Yes, move the set outside this function if the function is called multiple times